draft-rescorla-tls-esni.md

```diff
@@ -402,20 +396,31 @@ serves the connection directly (if in Shared Mode), in which case
 it executes the steps in the following section, or forwards
 the TLS connection to the hidden server (if in Split Mode).
 
+## Shared Mode Hidden Server Behavior
 
-## Hidden Server Behavior
+A server operating in Shared Mode uses PaddedServerNameList.sni as
+if it were the "server_name" extension to finish the handshake. It
+SHOULD pad the Certificate message, via padding at the record layer,
+such that its length equals the size of the largest possible Certificate 
+(message) covered by the same ESNI key.
+
+## Split Mode Hidden Server Behavior {#hidden-server-behavior}
 
 The Hidden Server ignores both the "encrypted_server_name" and the
 "server_name" (if any) and completes the handshake as usual. If in
 Shared Mode, the server will still know the true SNI, and can use it
 for certificate selection. In Split Mode, it may not know the true
-SNI and so will generally be configured to use a single certificate
+SNI and so will generally be configured to use a single certificate.
 {{communicating-sni}} describes a mechanism for communicating the
 true SNI to the hidden server.
-[[OPEN ISSUE: Do we want "encrypted_server_name" in EE? It's
-clearer communication, but gets in the way of stock servers.]]
 
+Similar to the Shared Mode behavior, the hidden server in Split Mode
+SHOULD pad the Certificate message, via padding at the record layer
+such that its length equals the size of the largest possible Certificate
+(message) covered by the same ESNI key.
```

Comment:
Again, this recommendation describes a single anonymity set.  Because the size of esni and certificate both contribute to information that distinguishes connections, you should include text on padding that addresses the combination of the two fields.

Reply:
Yep, good point. I’ll update in a separate PR.